qa/setup_packages.py

```diff
@@ -60,20 +85,26 @@ def get_key_with_cuda(key, val_dict, cuda):
     return key_w_cuda, max_cuda
 
 def get_package(package_data, key, cuda):
+    """Returns a list of available versions for given package name and cuda version"""
+    ret = None
     if key in package_data.keys():
         if isinstance(package_data[key], dict):
             data = None
             for ver in sorted(package_data[key].keys(), key=int):
                 if int(ver) <= int(cuda):
                    data = package_data[key][ver]
-            return data
+            ret = data
         else:
-            return packages[key]
-    else:
-        return None
+            ret = packages[key]
+
+    return filter_packages(key, ret)
 
 def test_request(req, name, cuda):
+    """Check if given url to a package is avialable"""
```

Comment:
```suggestion
def test_request(ver, url, cuda):
    """Check if given `url` to a package of version `ver` is avialable"""
```
If you are already touching this, but I won't insist.

Reply:
Done